src/libstd/sys/unix/thread.rs

```diff
@@ -52,6 +52,11 @@ impl Thread {
         assert_eq!(libc::pthread_attr_init(&mut attr), 0);
 
         let stack_size = cmp::max(stack, min_stack_size(&attr));
+
+        // L4Re only supports a maximum of 1Mb per default.
+        #[cfg(target_os = "l4re")]
+        let stack_size = cmp::min(stack_size, 1024 * 1024);
```

Comment:
The standard library tends to prefer to expose the underlying platform semantics as much as possible, avoiding trying to make shims and such where possible. It sounds like the default stack size on l4re should be less, but I believe we should still let the error propagate from `pthread_attr_setstacksize`